src/Doctrine/ODM/MongoDB/Aggregation/VectorSearchStage.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Doctrine\ODM\MongoDB\Aggregation;
+
+use Doctrine\ODM\MongoDB\Aggregation\Stage;
+
+final class VectorSearchStage extends Stage
```

Comment:
Awesome - we should add that stage, along with docs and examples for vector search etc. as part of our AI goal.

Reply:
[PHPORM-346](https://jira.mongodb.org/browse/PHPORM-346)